### PR TITLE
Updated authentication doc to match latest API

### PR DESCRIPTION
**Pull request by NSeydoux**
- Use the simplified `Session` constructor
- Call `handleIncomingRedirect` before checking if logged in
- No longer use URL objects
- Added explanations in the doc about why a redirection is necessary.

### CODE REVIEW

File: source/tutorial/authenticate.rst [hunk 1]
```diff
@@ -11,7 +11,21 @@ access </tutorial/manage-access-control-list>` to that data. For
...
+   account),
+
+#. Log in the Identity Provider
+
+#. Be redirected back to your app.
+
+The login is only complete after the last step, which means your app must be 
+reloaded between the moment when you initiate the login, and the moment you 
+complete it. It's important to keep this in mind to understand the way 
+``solid-client-authn-browser`` works.
```
Comment (kay-kim) [inline source/tutorial/authenticate.rst:28]: Maybe
```suggestion
.. important::
   With ``solid-client-authn-browser``, the login is only complete after the last step; i.e., your application must be reloaded between the start of the login and its completion. 
```

**Review by kay-kim — APPROVED** [review]
lgtm mod small suggestion

**Review by Vinnl — APPROVED** [review]
I think there's some overlap with my PR, but looks good :)

**Comment by NSeydoux** [comment]
@Vinnl I resolved the conflicts with https://github.com/inrupt/docs-client-lib-js/pull/31, because indeed we had introduced similar fixes to the same file :)

**Pull request merged**